### ISSUE TITLE
5x regression in `flutter_gallery_macos__start_up`:`timeToFirstFrameRasterizedMicros`

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#144341]
First seen in roll: #144288

[fa322eb295b69a7f76c32460b8e5f550ed12d880](https://github.com/flutter/flutter/commit/fa322eb295b69a7f76c32460b8e5f550ed12d880)

Probably related to: https://github.com/flutter/engine/pull/49159

[skia perf benchmark](https://flutter-flutter-perf.skia.org/e/?keys=Xbf8dce9c233bce34d20e2280e3f1d8db&selected=commit%3D39569%26name%3D%252Carch%253Dintel%252Cbranch%253Dmaster%252Cconfig%253Ddefault%252Cdevice_type%253Dnone%252Cdevice_version%253Dnone%252Chost_type%253Dmac%252Csub_result%253DtimeToFirstFrameRasterizedMicros%252Ctest%253Dflutter_gallery_macos__start_up%252C&xbaroffset=39569)

cc @cbracken

**Comment by cbracken** [comment]
@gaaclarke did other benchmarks regress in tandem with this or is it just the first frame latency?

**Comment by cbracken** [comment]
Answer from @gaaclarke over chat was that it was just the macOS first-frame benchmark. We didn't see any significant regressions on other macOS benchmarks.

**Comment by knopp** [comment]
For initial frame the vsync waiter currently waits for CVDisplayLink callback. CVDisplayLink start -> callback has rather big latency, because the callback gets invoked 1.5 frame durations before target time. Which means it skips whole extra frame compared to fallback waiter.

This is not a problem for subsequent frames because after a callback we remember the phase and when unpausing paused DisplayLink will schedule first callback using timer.

I'll take better look at this tomorrow to make sure I'm not missing anything, but we should be able to signal the warm-up frame without waiting for actual vsync.

**Comment by knopp** [comment]
I don't seem to be able to reproduce this locally. I'm getting fairly consistent `timeToFirstFrameRasterizedMicros` time whether I tried it with current main or revision before the PR rolled in. Also with local engine it does not matter whether I revert the CVDisplayLink PR or not.

I made PR for speculative fix https://github.com/flutter/engine/pull/51092. It bypasses the waiting for vsync for first frame and improves the `timeToFirstFrameRasterizedMicros` when I artificially delay the vsync callback.
